src/ViewHelpers/ArgumentViewHelper.php

```diff
@@ -0,0 +1,68 @@
+<?php
+namespace TYPO3Fluid\Fluid\ViewHelpers;
+
+/*
+ * This file belongs to the package "TYPO3 Fluid".
+ * See LICENSE.txt that was shipped with this package.
+ */
+
+use TYPO3Fluid\Fluid\Core\Rendering\RenderingContextInterface;
+use TYPO3Fluid\Fluid\Core\ViewHelper\AbstractViewHelper;
+use TYPO3Fluid\Fluid\Core\ViewHelper\Traits\CompileWithContentArgumentAndRenderStatic;
+
+/**
+ * Argument assigning ViewHelper
+ *
+ * Assigns an argument for a parent ViewHelper call when
+ * the parent ViewHelper supports it.
+ *
+ * Alternative to declaring an array to pass as "arguments".
+ *
+ * Usages:
+ *
+ *     <f:render partial="Foo">
+ *         <f:argument name="arg1">Value1</f:argument>
+ *         <f:argument name="arg2">Value2</f:argument>
+ *     </f:render>
+ *
+ * Which is the equivalent of:
+ *
+ *     <f:render partial="Foo" arguments="{arg1: 'Value1', arg2: 'Value2'}'" />
+ *
+ * But has the benefit that writing ViewHelper expressions or
+ * other more complex syntax becomes much easier because you
+ * can use tag syntax (tag content becomes argument value).
+ *
+ * @api
+ */
+class ArgumentViewHelper extends AbstractViewHelper
+{
+    use CompileWithContentArgumentAndRenderStatic;
+
+    /**
+     * @return void
+     */
+    public function initializeArguments()
+    {
+        $this->registerArgument('value', 'mixed', 'Value to assign. If not in arguments then taken from tag content');
```

Comment:
Should we define `$contentArgumentName` instead as suggested by the `CompileWithContentArgumentAndRenderStatic` trait?

Reply:
That hint is currently not valid as you cannot override properties declared in traits. The larger plan is to refactor content argument support into the base VH class.

Reply:
You can set a value in the constructor though.

Reply:
This is true but that is not good practice either. We really shouldn't use a new solution here if we expect to refactor it into the base class - since the solution could very easily be incompatible. Don't forget that internally Fluid will just use the first argument we declare.